### DISCUSSION TITLE
custom routing in Go?

### DISCUSSION THREAD

**Discussion by astroposyt** [pocketbase/pocketbase#5291] [Q&A] [answered]
I am currently trying to setup my pocketbase backend to listen for webhooks from paddle. I am trying to follow the extend with go docs to setup just a custom route, but I am getting no where with it.

Here is the code I am trying:

```GO
func main() {
    app := pocketbase.New()
	
    if err := app.Start(); err != nil {
        log.Fatal(err)
    }
	
    // serves static files from the provided public dir (if exists)
    app.OnBeforeServe().Add(func(e *core.ServeEvent) error {
		
		e.Router.GET("/hello/:name", func(c echo.Context) error {
			fmt.Printf("hello world!")
			name := c.PathParam("name")
	
			return c.JSON(http.StatusOK, map[string]string{"message": "Hello " + name})
		}, /* optional middlewares */)
	
		return nil
	})




}
```
When I send a curl to this `curl http://localhost:8090/hello/a ` I get an error  `Not Found.
   └─ code=404, message=Not Found`

What can i do to get a route for paddles webhooks

**Comment by ganigeorgiev** [comment, marked as answer]
You need to register the hooks **before** the `app.Start()` call (see the example in https://pocketbase.io/docs/go-overview/).

**Reply by astroposyt** [reply]
Amazing, it is working now. Can't believe I spent so long on that

**Reply by astroposyt** [reply]
Do you know why this doesn't(wouldn't) work on the deployed version in fly.io?

**Reply by ganigeorgiev** [reply]
In general it depends how you are deploying and building your Dockerfile.

It is possible that you are pushing an old image (aka. you may have to run docker build) or maybe you are not copying correctly the changed files in your image or if you are building locally and your image contains just the resulting binary then you may have to run `go build`, etc. There are just too many missing details to guess what could be the issue.
